### DISCUSSION TITLE
How can I serve superset under reverse proxy custom routes?

### DISCUSSION THREAD

**Discussion by BRAVO68WEB** [apache/superset#23156] [Ideas]
We want to run all superset services under http://host/analytics. But as of right now, all requests are redirected to http://host/superset. Require assistance on this.

**Comment by BRAVO68WEB** [comment]
I am facing an issue

curl http://localhost/superset ->  http://localhost/dashboard/list
it should be http://localhost/superset/dashboard/list

curl http://localhost/superset/dashboard/list -> http://localhost/dashboard/list
it should be http://localhost/superset/dashboard/list

I am running apache superset at http://localhost/superset as reverse proxy on nginx

**Reply by Victortehgk** [reply]
Facing the same issue, only workaround I've found is only this [https://github.com/komoot/superset-reverse-nginx-example/tree/master?tab=readme-ov-file ](url)

**Comment by Jbaudon** [comment]
I'm facing the same issue, I tried to implement several workaround and failed to correctly rewrite every paths, especially API calls.
The best option I've found is to add a Flask Middleware:
```python
class PrefixMiddleware(object):
  def __init__(self, app, prefix='/superset'):
    self.app = app
    self.prefix = prefix
  def __call__(self, environ, start_response):
    if environ['PATH_INFO'].startswith(self.prefix):
      environ['PATH_INFO'] = environ['PATH_INFO'][len(self.prefix):]
      environ['SCRIPT_NAME'] = self.prefix
      return self.app(environ, start_response)
    else:
      start_response('404', [('Content-Type', 'text/plain')])
      return ["This url does not belong to the app.".encode()]
ADDITIONAL_MIDDLEWARE = [PrefixMiddleware, ]
```
I also rewrote some hard-coded paths on reverse-proxy side but I failed to rewrite hard-coded paths included in the javascript assets.

**Comment by glego** [comment]
Since I am also interested in this feature, I have included links to all relevant issues, pull requests, and discussions. So far, it appears that only subdomains are supported with Nginx. For more details, see this comment:  https://github.com/apache/superset/pull/1866#issuecomment-335272079 and https://github.com/apache/superset/pull/1866#issuecomment-347310860

Related Issues:
https://github.com/apache/superset/issues/22842
https://github.com/apache/superset/issues/22850
https://github.com/apache/superset/issues/3690
https://github.com/apache/superset/issues/985

Related Pull Requests:
https://github.com/apache/superset/pull/1866

Related Discussions:
https://github.com/apache/superset/discussions/14440
